var/spack/repos/builtin/packages/mongo-c-driver/package.py

```diff
@@ -45,16 +51,32 @@ class MongoCDriver(Package):
         when="@1.8.1",
     )
 
-    depends_on("cmake@3.1:", type="build", when="@1.10.0:")
+    with when("build_system=cmake"):
+        depends_on("cmake@3.1:", type="build")
+
+    with when("build_system=autotools"):
+        depends_on("autoconf", type="build", when="@1.8.1")
+        depends_on("automake", type="build", when="@1.8.1")
+        depends_on("libtool", type="build", when="@1.8.1")
```

Comment:
Is m4 missing here on purpose?

Reply:
Yeah, m4 is a run dependency of one of those and not used directly